README.md
# QR Code Generator
Generate your own QR codes!

Steps:


[EN]
1. Fill all the fields
Add the url to be redirected
Add the name of the png file that will contain the QR code
Add an integer value to SCALE property. I recommend you to use 5.

You can use values from 1 to N.

The recommended scale values are from 1 to 10

2. Go to TOOLS > Generate QR Code

3. You'll see the messagebox tellling you how the QR code was saved.

The PNG file of the QR code will be saved inside the same directory as the .exe file.

[ES]
1. Rellena todos los campos
Agregue la URL para ser redirigido
Agregue el nombre del archivo png que contendrá el código QR
Agregue un valor entero a la propiedad SCALE. Te recomiendo que uses 5.

Puede utilizar valores del 1 a N.

Los valores de escala recomendados son del 1 al 10

2. Vaya a HERRAMIENTAS > Generar código QR

3. Verá el cuadro de mensaje que le indica cómo se guardó el código QR.

El archivo PNG del código QR se guardará dentro del mismo directorio que el archivo .exe.

![image](https://user-images.githubusercontent.com/89426850/176270287-036ebe2f-48dc-4d11-9715-832c95eb24e6.png)


![image](https://user-images.githubusercontent.com/89426850/176270368-8b2f4496-5ad1-4afc-ad1c-907b225accaf.png)


![image](https://user-images.githubusercontent.com/89426850/176271295-d3027787-571f-4cb5-9d5f-206a94819ec3.png)


![image](https://user-images.githubusercontent.com/89426850/176271301-8d9964c9-ad53-4356-86a0-291cb8236533.png)



[EN]
To download go to COMPILED > ENGLISH > QRCode.exe > View Raw

Extra: THe "RESET" button is used to clean the text fields.

(This program was compiled only for Windows. If you want to run it on MacOs or Linux, you'll need to run it from the source code)

[ES]
Para descargar vaya a COMPILED > SPANISH > QRCode_ES.exe > View Raw

Extra: El botón "LIMPIAR" se utiliza para dejar en blanco los campos de texto.

(Este programa fue compilado solo para Windows. Si desea ejecutarlo en MacOs o Linux, deberá ejecutarlo desde el código fuente)


CHANGELOG.txt
1. Added a Spanish version of the program

src/app_es.py
from tkinter import *
from tkinter import ttk
from tkinter import messagebox
from tkinter import filedialog

from uuid import uuid4

import png
import pyqrcode 

from time import sleep

class App:
	def __init__(self):
		self.app = Tk()
		self.app.title("Generador de códigos QR")
		self.app.geometry('430x200')
		self.app.resizable(False,False)
		self.app['bg'] = 'teal'

		# menu
		mainmenu = Menu(self.app)

		toolsmenu = Menu(mainmenu, tearoff=0)
		toolsmenu.add_command(label="Generar Código QR", command=self.generate_code)

		mainmenu.add_cascade(label="Herramientas", menu=toolsmenu)
		mainmenu.add_command(label="Limpiar", command=self.reset_values)
		mainmenu.add_command(label="Salir", command=self.exit_app)

		self.app.config(menu=mainmenu)

		# url entry
		url_lbl = Label(self.app, text="ENLACE DE DESTINO:", font="bold 12", fg="white")
		url_lbl.place(x=10,y=7)
		url_lbl['bg'] = 'teal'

		self.urlVar = StringVar()
		url_entry = Entry(self.app, width=65, textvariable=self.urlVar)
		url_entry.place(x=10,y=30)

		# image config
		name_lbl = Label(self.app, text="NOMBRE DEL ARCHIVO PNG:", font="bold 12", fg='white')
		name_lbl.place(x=10,y=65)
		name_lbl['bg'] = 'teal'

		self.nameVar = StringVar()
		url_entry = Entry(self.app, width=65, textvariable=self.nameVar)
		url_entry.place(x=10,y=90)

		# scale config
		name_lbl = Label(self.app, text="ESCALA DE LA IMAGEN\n(Valores recomendables: Enteros del 1 al 10)", font="bold 12", fg='white')
		name_lbl.place(x=10,y=125)
		name_lbl['bg'] = 'teal'

		self.scaleVar = IntVar()
		url_entry = Entry(self.app, width=30, textvariable=self.scaleVar)
		url_entry.place(x=10,y=170)


	def exit_app(self):
		self.app.quit()

	def reset_values(self):
		self.urlVar.set(" ")
		self.nameVar.set(" ")
		self.scaleVar.set(0)

	def generate_code(self):
		url = self.urlVar.get()
		name = self.nameVar.get()
		
		try:
			scale = self.scaleVar.get()
			scale = int(scale)
			if(len(url) > 0 and len(name) > 0 and scale > 0):
				try:
					code = pyqrcode.create(url)
					filename = f"{name.upper()}_{str(uuid4())}.png"
					with open(filename, 'wb') as qr:
						code.png(qr, scale=scale)
					messagebox.showinfo("Generador de códigos QR", "Archivo guardado como {}".format(filename))
				except:
					messagebox.showerror("Error", "Ha ocurrido un error")
			else:
				messagebox.showerror("Error", "Ingresa una URL de destino y un nombre para el archivo PNG\nLa escala debe ser > 0")
		except:
			messagebox.showerror("Error", "Solo valores numericos")

	def run(self):
		self.app.mainloop()

if __name__=="__main__":
	GUI = App()
	GUI.run()

src/app.py
from tkinter import *
from tkinter import ttk
from tkinter import messagebox
from tkinter import filedialog

from uuid import uuid4

import png
import pyqrcode 

from time import sleep

class App:
	def __init__(self):
		self.app = Tk()
		self.app.title("QR Code Generator")
		self.app.geometry('430x200')
		self.app.resizable(False,False)
		self.app['bg'] = 'teal'

		# menu
		mainmenu = Menu(self.app)

		toolsmenu = Menu(mainmenu, tearoff=0)
		toolsmenu.add_command(label="Generate QR Code", command=self.generate_code)

		mainmenu.add_cascade(label="Tools", menu=toolsmenu)
		mainmenu.add_command(label="Reset", command=self.reset_values)
		mainmenu.add_command(label="Exit", command=self.exit_app)

		self.app.config(menu=mainmenu)

		# url entry
		url_lbl = Label(self.app, text="URL:", font="bold 12", fg="white")
		url_lbl.place(x=10,y=7)
		url_lbl['bg'] = 'teal'

		self.urlVar = StringVar()
		url_entry = Entry(self.app, width=65, textvariable=self.urlVar)
		url_entry.place(x=10,y=30)

		# image config
		name_lbl = Label(self.app, text="File Name:", font="bold 12", fg='white')
		name_lbl.place(x=10,y=65)
		name_lbl['bg'] = 'teal'

		self.nameVar = StringVar()
		url_entry = Entry(self.app, width=65, textvariable=self.nameVar)
		url_entry.place(x=10,y=90)

		# scale config
		name_lbl = Label(self.app, text="File Scale (Integer value)", font="bold 12", fg='white')
		name_lbl.place(x=10,y=125)
		name_lbl['bg'] = 'teal'

		self.scaleVar = IntVar()
		url_entry = Entry(self.app, width=30, textvariable=self.scaleVar)
		url_entry.place(x=10,y=150)


	def exit_app(self):
		self.app.quit()

	def reset_values(self):
		self.urlVar.set(" ")
		self.nameVar.set(" ")
		self.scaleVar.set(0)

	def generate_code(self):
		url = self.urlVar.get()
		name = self.nameVar.get()
		
		try:
			scale = self.scaleVar.get()
			scale = int(scale)
			if(len(url) > 0 and len(name) > 0 and scale > 0):
				try:
					code = pyqrcode.create(url)
					filename = f"{name.upper()}_{str(uuid4())}.png"
					with open(filename, 'wb') as qr:
						code.png(qr, scale=scale)
					messagebox.showinfo("QR Code Generator", "QR Code saved as {}".format(filename))
				except:
					messagebox.showerror("Error", "An error occurred")
			else:
				messagebox.showerror("Error", "Enter an URL and a file name.\nThe file scale must be > 0")
		except:
			messagebox.showerror("Error", "Integer values only")

	def run(self):
		self.app.mainloop()

if __name__=="__main__":
	GUI = App()
	GUI.run()